### PR TITLE
Change `arrowParens` to `always` by default for 2.0

### LINKED ISSUE #6929: Change `arrowParens` to `always` by default for 2.0
Why i think we need to change that?

- code consistency when an argument more than one
- if we look through our issues a lot of developers use `always`
- it does not require large code changes in our code
- honestly I don't even know why we once created this option and whether we need it in the future

Maybe we should create poll on twitter and here

@prettier/core I would like to hear your opinions on this

Ref: https://github.com/prettier/prettier/issues/6888

### PR DESCRIPTION
**Pull request by kachkaev**
Closes #6929

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: changelog_unreleased/api/pr-7430.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<!--
```
Comment (kachkaev) [inline changelog_unreleased/api/pr-7430.md]: This file is still a TODO. See https://github.com/prettier/prettier/pull/6963#issuecomment-578409884 for context.

**Comment by lipis** [comment]
Is that have been decided? :(

**Comment by kachkaev** [comment]
@lipis I haven't heard objections with many 👍 in reactions, so it seems like changing this default is a go. See https://github.com/prettier/prettier/issues/6929 and https://github.com/prettier/prettier/issues/6888#issuecomment-572570924 for details.

**Comment by lipis** [comment]
Whiile we at it.. let's remove the bracketSpacing too by default

**Comment by kachkaev** [comment]
Changing default for `bracketSpacing` would probably require a discussion like #6929, #6932 or #6931, which would be up for at least a couple of weeks. Not sure if there's enough time for this before 2.0 TBH 🤔

File: changelog_unreleased/api/pr-7430.md (hunk 1 above)
Comment (kachkaev) [reply changelog_unreleased/api/pr-7430.md]: This is done now. The whole PR is ready for a review. Once this is done, I'm happy to help @fisker with his portion of changelog (see https://github.com/prettier/prettier/pull/6963#issuecomment-578434520).

File: docs/options.md [hunk 2]
```diff
@@ -155,18 +155,18 @@ Valid options:
...
 Valid options:
 
-- `"avoid"` - Omit parens when possible. Example: `x => x`
 - `"always"` - Always include parens. Example: `(x) => x`
+- `"avoid"` - Omit parens when possible. Example: `x => x`
 
 | Default   | CLI Override                                    | API Override                                    |
 | --------- | ----------------------------------------------- | ----------------------------------------------- |
-| `"avoid"` | <code>--arrow-parens <avoid&#124;always></code> | <code>arrowParens: "<avoid&#124;always>"</code> |
+| `"avoid"` | <code>--arrow-parens <always&#124;avoid></code> | <code>arrowParens: "<always&#124;avoid>"</code> |
```
Comment (kachkaev) [inline docs/options.md]: Oops, forgot to change ``| `"avoid"` |`` to ``| `"always"` |``

Comment (kachkaev) [reply docs/options.md]: ✅

**Comment by kachkaev** [comment]
@ExE-Boss these concerns have been discussed  in #6929. A noise-free visual look of the code and a great developer experience are not always in line, the latter has higher priority. Parentheses are a similar story to semicolons or single / double quotes.

With semicolons, every line has a noisy character, but a developer does not have to think about horrible edge cases like `;[]` or `;()`. With double quotes, there's 50% more 'ink' used, but there is consistency between JS and other languages (including JSON), which improves DX in certain scenarios like search.

You don't have to agree with the benefits of consistency and [_rule shortness_](https://github.com/prettier/prettier/issues/6929#issuecomment-552646445) when making formatting decision for your projects. If that's the case, Prettier offers you configuration options. The ability to set `{ "arrowParens": "avoid" }` does not go anywhere in v2.0 😉

**Comment by karlhorky** [comment]
@kachkaev I put my 👍 on your comment as I think that the parallel with semicolons makes sense! There are footguns with no-semicolon JS, and there are almost-footguns with parentheses (while adding a new parameter afterwards).

However, I think the "double quotes" issue is different and the default should be changed here in the future to single quotes (see #4102). Why? Because it's not about other languages, it's about JavaScript. And JavaScript is much more commonly written with single quotes in the wild.

Plus, there are no footguns with single vs double. If you wanted to talk about footguns, actually the better option would be to switch everything to template strings, so that you can easily interpolate later (parallel: easily adding parameters later). But this style is even less common, and I would not suggest that Prettier change to this.

**Comment by evilebottnawi** [comment]
@kachkaev Can you rebase?

**Comment by kachkaev** [comment]
Oh, I did not notice there were conflicts, thanks for pointing this out @evilebottnawi  😬 I was a bit wondering: why does everyone approve the changes, but nobody merges? 😅

Instead of rebasing, I merged `next` branch into my branch and resolved conflicts in snapshots afterwards. The diff looks clean now. The PR will be squashed anyway, so the details of the commit history does not matter much. By not rebasing, I keep git history more 'honest' and thus easier to follow for those who are curios how this PR was emerging 🙂

**Comment by jurosh** [comment]
> I disagree with this as it makes code like in [#6929 (comment)](https://github.com/prettier/prettier/issues/6929#issuecomment-552637754) much noisier:
> 
> > `arrowParens: "avoid"`: 👍🏻
> > 
> > ```js
> > categories
> > 	.filter(x => x.name === category)
> > 	.flatMap(x => x.pages)
> > 	.filter(x => x.name.includes(search))
> > 	.forEach(x => (x.highlight = true));
> > ```
> > 
> > 
> > 
> > `arrowParens: "always"`: 👎🏻
> > 
> > ```js
> > categories
> > 	.filter((x) => x.name === category)
> > 	.flatMap((x) => x.pages)
> > 	.filter((x) => x.name.includes(search))
> > 	.forEach((x) => (x.highlight = true));
> > ```

Agree with @ExE-Boss 
For functional programming maps, filters this looks much worse. For typescript, this would make sense only for named arrow functions - BUT never for unnamed functions in maps/promises.

So I am pretty sad by this new default as there will be lot of new fights in development teams to keep using this new or always use `arrowParens: "avoid"`. I would def use `avoid` whenever possible.

**Pull request merged**